packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -16,101 +23,165 @@ interface ScheduledAppointmentsProps {
+  const shouldShowPanel = useCallback(
+    (panel: Omit<ConnectedExtension, 'config'>) => allowedExtensions[panel.name] ?? false,
+    [allowedExtensions],
+  );
 
   useEffect(() => {
-    if (dayjs(date).isBefore(dayjs(), 'date')) {
+    const dayjsDate = dayjs(date);
+    const now = dayjs();
+    if (dayjsDate.isBefore(now, 'date')) {
       setDateType('pastDate');
-    } else if (dayjs(date).isAfter(dayjs(), 'date')) {
+    } else if (dayjsDate.isAfter(now, 'date')) {
       setDateType('futureDate');
     } else {
       setDateType('today');
     }
   }, [date]);
 
-  // set the default tab on first render
   useEffect(() => {
-    if (!currentTab) {
-      setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
+    if (currentTab === null && allowedExtensions) {
+      for (const extension of Object.getOwnPropertyNames(allowedExtensions)) {
+        if (allowedExtensions[extension]) {
+          setCurrentTab(extension);
+          break;
+        }
+      }
     }
-  }, [scheduledAppointmentPanels]);
-
-  // set the default tab whenever the date type changes
-  useEffect(() => {
-    setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
-  }, [dateType]);
-
-  const getDefaultTab = (panels) => {
-    return panels?.find(
-      (panel) =>
-        (isDateInPast && panel.config.showForPastDate) ||
-        (isToday && panel.config.showForToday) ||
-        (isDateInFuture && panel.config.showForFutureDate),
-    )?.config.title;
-  };
+  }, [allowedExtensions, currentTab]);
 
   return (
     <>
-      {dateType === 'today' && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
-          {scheduledAppointmentPanels
-            .filter((panel) => panel.config.showForToday)
-            .map((panel, i) => {
-              return <Switch key={i} name={panel.config.title} text={t(panel.config.title)} />;
-            })}
-        </ContentSwitcher>
-      )}
-
-      {dateType === 'pastDate' && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
-          {scheduledAppointmentPanels
-            .filter((panel) => panel.config.showForPastDate)
-            .map((panel, i) => {
-              return <Switch key={i} name={panel.config.title} text={t(panel.config.title)} />;
-            })}
-        </ContentSwitcher>
-      )}
-
-      {dateType === 'futureDate' && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
-          {scheduledAppointmentPanels
-            .filter((panel) => panel.config.showForFutureDate)
-            .map((panel, i) => {
-              return <Switch key={i} name={panel.config.title} text={t(panel.config.title)} />;
-            })}
-        </ContentSwitcher>
-      )}
+      <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
```

Comment:
I collapsed the various `ContentSwitcher`s into a single `ContentSwitcher` filtered using the `shouldShowPanel()` callback defined above. I also used a key that's derived from the extension name, as that's a more stable identifier and it's always best to use the data over a simple counter if possible because it helps reduce React re-renders.